### DISCUSSION TITLE
Don't show current commit in smartlog

### DISCUSSION THREAD

**Discussion by adzenith** [arxanas/git-branchless#1240] [Q&A] [answered]
I want to use `git smartlog` to show a chunk of the commit graph. It seems that if I provide a query, it also includes the currently checked out commit even if the query wouldn't otherwise include that commit. Is there a way to disable this behavior?
The help does promise:
```
  [REVSET]  The commits to render. These commits, plus any related commits, will be rendered
```
so perhaps this is a "related commit" :) But is there a way I can have it show just my revset?

Thanks!

**Comment by claytonrcarter** [comment, marked as answer]
> also includes the currently checked out commit even if the query wouldn't otherwise include that commit

This was intentional, as the smartlog is primarily a situational awareness tool rather than a query or reporting tool. There are 2 commits always included to assist with understanding how the revset relates to the public commits (`main` or `master` or whatever) and to what you're currently working on (`HEAD`).

That said, it would not be hard to add a flag to turn this off. The code that adds these commits to the log is here:
https://github.com/arxanas/git-branchless/blob/c9dcc0077f00b5f5b718bb14dd6bb1b220bcfd34/git-branchless-smartlog/src/lib.rs#L350-L353

**Comment by adzenith** [comment]
An `--exact` flag was added in https://github.com/arxanas/git-branchless/pull/1241
